pkg/manager/mce.go

```diff
@@ -0,0 +1,373 @@
+			GCP: &hivegcp.Platform{
+				CredentialsSecretRef: v1.LocalObjectReference{
+					Name: fmt.Sprintf("%s-%s-creds", clusterName, platform),
+				},
+				Region: "us-east1",
+			},
+		}
+	}
+
+	if err := m.dpcrHiveClient.Create(context.TODO(), &clusterDeployment, &client.CreateOptions{}); err != nil {
+		return nil, fmt.Errorf("failed to create cluster deployment: %v", err)
+	}
+
+	return &managedCluster, nil
+}
+
+func (m *jobManager) listManagedClusters() ([]*clusterv1.ManagedCluster, []*hivev1.ClusterDeployment, error) {
+	namespaces, err := m.dpcrNamespaceClient.List(context.TODO(), metav1.ListOptions{LabelSelector: utils.LaunchLabel})
+	if err != nil {
+		return nil, nil, fmt.Errorf("Failed to get list of managed clusters: %v", err)
+	}
+	managedClusters := []*clusterv1.ManagedCluster{}
+	clusterDeployments := []*hivev1.ClusterDeployment{}
+	for _, namespace := range namespaces.Items {
+		cluster := clusterv1.ManagedCluster{}
+		if err := m.dpcrOcmClient.Get(context.TODO(), client.ObjectKey{Namespace: namespace.Name, Name: namespace.Name}, &cluster); err != nil {
+			klog.Warningf("Failed to get managed cluster %s; cluster is likely being deleted: %v", namespace.Name, err)
+		} else {
+			managedClusters = append(managedClusters, &cluster)
+		}
+		clusterDeployment := hivev1.ClusterDeployment{}
+		if err := m.dpcrHiveClient.Get(context.TODO(), client.ObjectKey{Namespace: namespace.Name, Name: namespace.Name}, &clusterDeployment); err != nil {
+			klog.Warningf("Failed to get managed cluster %s; cluster is likely being deleted: %v", namespace.Name, err)
+		} else {
+			clusterDeployments = append(clusterDeployments, &clusterDeployment)
+		}
+	}
+	return managedClusters, clusterDeployments, nil
+}
+
+// getClusterAuth returns kubeconfig,password,error
+func (m *jobManager) getClusterAuth(name string) (string, string, error) {
+	m.mceClusters.lock.RLock()
+	// don't pull clusterdeployment if auth is already cached
+	existingKubeconfig, ok := m.mceClusters.clusterKubeconfigs[name]
+	existingPassword, ok2 := m.mceClusters.clusterPasswords[name]
+	m.mceClusters.lock.RUnlock()
+	if ok && ok2 && existingKubeconfig != "" && existingPassword != "" {
+		return existingKubeconfig, existingPassword, nil
+	}
+	clusterDeployment := hivev1.ClusterDeployment{}
+	if err := m.dpcrHiveClient.Get(context.TODO(), client.ObjectKey{Namespace: name, Name: name}, &clusterDeployment); err != nil {
+		return "", "", err
+	}
+	if clusterDeployment.Spec.ClusterMetadata == nil {
+		return "", "", nil
+	}
+	secretsClient := m.dpcrCoreClient.Secrets(name)
+	kubeconfigSecret, err := secretsClient.Get(context.TODO(), clusterDeployment.Spec.ClusterMetadata.AdminKubeconfigSecretRef.Name, metav1.GetOptions{})
+	if err != nil {
+		return "", "", err
+	}
+	/*
+		kubeconfig, err := base64.RawStdEncoding.DecodeString(string(kubeconfigSecret.Data["kubeconfig"]))
+		if err != nil {
+			return "", "", fmt.Errorf("failed to decode kubeconfig: %v", err)
+		}
+	*/
```

Comment:
Just want to make sure we don't need the decoded strings here?  Can probably be deleted if not.

Reply:
We don't need to decode the strings. I've removed this code